### ISSUE TITLE
Improve readability of diagnotic hovers

### ISSUE THREAD

**Issue by felixfbecker** [Microsoft/vscode#54503]
The number one annoyance I have and hear from people about VS Code + TypeScript is that type errors are hard to read. TypeScript will display nested type errors with each level indented, but because of the line wrapping, that indentation basically lost.
This makes the error hard to follow and in consequence hard to solve.

![image](https://user-images.githubusercontent.com/10532611/42839855-dbc732a2-89b9-11e8-841d-f0b66f3a62fb.png)

Here are things that I think would improve the readability:
- Add an option to enable horizontal scrolling in hovers. This would mean the indentation would actually have the intended visual effect.
- Make the width of the hover configurable. I have way more space on my screen it could utilize.
- Add smart wrapping that preserves indentation.
- Add markdown support to diagnostics and make TypeScript output markdown. This would mean not the whole diagnostic has to be rendered in a monospace font, therefor saving horizontal space. The parts that are code would use backticks instead of single quotes.

**Comment by joaomoreno** [comment]
Hover has full Markdown support. TypeScript opts to spit everything out in monospace font, so it ends up like that.

**Comment by mjbvz** [comment]
@joaomoreno This is coming from VS Code's built-in error hover support which I do not believe supports markdown. Even if problem hover did support markdown content, wider hovers content and proper wrapping would still be beneficial and cannot be addressed by the TypeScript extension

**Comment by mjbvz** [comment]
Expanding hovers was added with #67076
